### PR TITLE
Deleting an n use image should return conflict not system error

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/14208

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: test/apiv2/10-images.at [hunk 1]
```diff
@@ -156,6 +156,17 @@ t GET images/json?filters='{"reference":["test1"]}' 200 length=1
 
 t DELETE libpod/images/test1:latest 200
 
+# to be used in prune until filter tests
+podman image build -t test1:latest -<<EOF
+from alpine
+RUN >file4
+EOF
+podman create test1 echo hi
```
Comment (vrothberg) [inline test/apiv2/10-images.at:164]:
```suggestion
podman create --name=test1 test1 echo hi
```

**Review by vrothberg — COMMENTED** [review]
Code LGTM
Test needs a massage

File: test/apiv2/10-images.at [hunk 2]
```diff
@@ -156,6 +156,17 @@ t GET images/json?filters='{"reference":["test1"]}' 200 length=1
...
 t DELETE libpod/images/test1:latest 200
 
+# to be used in prune until filter tests
+podman image build -t test1:latest -<<EOF
+from alpine
+RUN >file4
+EOF
+podman create --name test1 test1 echo hi
+
+t DELETE libpod/images/test1:latest 409
```
Comment (jwhonce) [inline test/apiv2/10-images.at:166]: The change is in the compat handler while the test is calling the libpod handler.
```suggestion
t DELETE images/test1:latest 409
```

File: test/apiv2/10-images.at [hunk 3]
```diff
@@ -156,6 +156,17 @@ t GET images/json?filters='{"reference":["test1"]}' 200 length=1
...
+podman rm test1
+t DELETE libpod/images/test1:latest 200
```
Comment (jwhonce) [inline test/apiv2/10-images.at:168]:
```suggestion
t DELETE images/test1:latest 200
```

**Review by jwhonce — CHANGES_REQUESTED** [review]
Test not covering new code.

**Review by vrothberg — COMMENTED** [review]
LGTM
@jwhonce PTanotherL

**Pull request merged**